files/generate_system_metrics

```diff
@@ -113,35 +124,46 @@ module SystemMetrics
       data_hash.transform_values!(&:to_f)
     end
 
-    # Convert the inputted sar output into json and raise errors if the sar output is invalid
+    # Create the file and put the json data with the error in it
+    # then exit
     #
     # @author Randell Pelak
     #
-    # @param [string] sar_output
-    # @param [obj] time_stamp_obj Time object to use for generating the filename
+    # @param [string] error_msg Error message to put in the json payload
     #
-    # @raise [RuntimeError] if sar_output doesn't parse correctly
+    # @return [void]
+    def send_error_to_output_file_and_exit(error_msg)
+      json_payload = create_json_payload({:error => error_msg})
+      create_file(json_payload)
+      exit 1
+    end
+
+    # Create json payload
     #
-    # @return [string] json of sar output
-    def convert_sar_output_to_json(sar_output, time_stamp_obj)
+    # @author Randell Pelak
+    #
+    # @param [hash] metrics_data the data for the metrics section of the jason
+    #
+    # @return [string] json payload
+    def create_json_payload(metrics_data)
```

Comment:
:pick: I preferred the language of the previous method.  Perhaps this could be called `metrics_to_json` or something else that indicates what the intended input is?

Reply:
works for me.